### PR TITLE
feat: add support for pydantic custom classes as fields

### PR DESCRIPTION
**Pull request by OffByOnee**
This change adds support for [Pydantic Custom Classes](https://docs.pydantic.dev/1.10/usage/types/#classes-with-__get_validators__) as fields and resolves [issue #392](https://github.com/marcosschroh/dataclasses-avroschema/issues/392). It updates the field factory to reference the `json_encoders` specified in Pydantic's Config when building the schema for a custom class (defined as a Python class containing the `validate` and `__get_validators__` class methods). It also updates `AvroBaseModel` serialization to leverage `json_encoders` for standardization prior to passing the data to the general Avro standardization function, so that the serialized data matches the schema.

Additionally it adds an extra level of validation to one of the Pydantic serialization tests (let me know if that should be reverted).

### CODE REVIEW

File: tests/serialization/test_pydantic_fields.py [hunk 1]
```diff
@@ -27,3 +63,22 @@ class ConstrainedType(AvroBaseModel):
 
     with pytest.raises(ValidationError):
         ConstrainedType.deserialize(b'{"value": 0}', serialization_type="avro-json")
+
+
+def test_generic_class_type_serialize():
```
Comment (marcosschroh) [inline tests/serialization/test_pydantic_fields.py:68]: Could you add a `parametrize` to test the `avro` serialization as well? `@pytest.mark.parametrize(
    "serialization_type", [AVRO, AVRO_JSON,]`

**Review by marcosschroh — COMMENTED** [review]
Could you add the documentation for `Pydantic Generic Classes` under the [pydantic section](https://marcosschroh.github.io/dataclasses-avroschema/pydantic/)? 

PS: We should mention as a `warning` that it is not possible to generate `models` from `avro schemas` that were generated from classes that contain Generics. It would be nice to have but they have so much custom and business logic so is not possible

**Comment by OffByOnee** [comment]
> Could you add the documentation for `Pydantic Generic Classes` under the [pydantic section](https://marcosschroh.github.io/dataclasses-avroschema/pydantic/)?
> 
> PS: We should mention as a `warning` that it is not possible to generate `models` from `avro schemas` that were generated from classes that contain Generics. It would be nice to have but they have so much custom and business logic so is not possible

Sure thing 👍

File: tests/serialization/test_pydantic_fields.py (hunk 1 above)
Comment (OffByOnee) [reply tests/serialization/test_pydantic_fields.py:68]: Absolutely

**Pull request merged**